src/application-navigation/ApplicationNavigation.jsx

```diff
@@ -0,0 +1,192 @@
+  /**
+   * Callback function triggered on Drawer Menu state change
+  */
+  onDrawerMenuStateChange: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of a navigation item.
+   * Ex: `onSelectNavigationItem(String selectedNavigationItemKey, Object metaData)`
+   */
+  onSelectNavigationItem: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Settings utility item.
+   * If `onSelectSettings` is not provided, the Settings utility item will not be rendered.
+   * Ex: `onSelectSettings()`
+   */
+  onSelectSettings: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Help utility item.
+   * If `onSelectHelp` is not provided, the Help utility item will not be rendered.
+   * Ex: `onSelectLogout()`
+   */
+  onSelectHelp: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Logout action button.
+   * If `onSelectLogout` is not provided, the Logout action button will not be rendered.
+   * Ex: `onSelectLogout()`
+   */
+  onSelectLogout: PropTypes.func,
+  /**
+   * An array of configuration objects with information specifying the creation of additional utility menu items.
+   * These items are rendered within the popup utility menu at larger breakpoints and within the drawer menu at smaller breakpoints.
+   */
+  utilityItems: utilityItemsPropType,
+  /**
+   * A function to be executed upon the selection of a custom utility item.
+   * Ex: `onSelectUtilityItem(String selectedUtilityItemKey, Object metaData)`
+   */
+  onSelectUtilityItem: PropTypes.func,
+  /**
+   * A collection of child elements to render within the ApplicationNavigation body.
+   */
+  children: PropTypes.node,
+  /**
+   * Key/Value pairs associating a string key entry to a numerical notification count.
+   */
+  notifications: PropTypes.object, // eslint-disable-line react/no-unused-prop-types
+  /**
+   * By default, the ApplicationNavigation component will resolve any rendered NavigationPrompts prior to
+   * communicating navigation item selections with `onSelectNavigationItem`. If `disablePromptsForNavigationItems`
+   * is provided, no NavigationPrompts are resolved when navigation items are selected.
+   */
+  disablePromptsForNavigationItems: PropTypes.bool,
+  /**
+   * By default, the ApplicationNavigation component will resolve any rendered NavigationPrompts prior to
+   * communicating logout selection with `onSelectLogout`. If `disablePromptsForLogout` is provided,
+   * no NavigationPrompts are resolved when logout is selected.
+   */
+  disablePromptsForLogout: PropTypes.bool,
+  /**
+   * The Object (or function that returns an Object) that specifies the messages
+   * used to prompt the user when navigation items are selected while NavigationPrompts
+   * are rendered by the ApplicationNavigation content.
+   */
+  navigationPromptResolutionOptions: navigationPromptResolutionOptionsShape,
+};
+
+const ApplicationNavigation = ({
+  children,
```

Comment:
nit: there are a lot of props for this component. can we try to order in ABC order to quickly find the prop defs?

Reply:
Yeah that's fair. How do we feel about the documented prop table order? I've always tended to order things in either logical groups or a rough order of "importance" (required or not).